### DISCUSSION TITLE
Support for generating "static" website by decoupling web server and api backend ?

### DISCUSSION THREAD

**Discussion by jcfr** [Kitware/trame#25] [Q&A]
Using python to create a website is great to enable researchers and scientists to easily put together beautiful web applications without having to learn typescript, ...

To allow serving the generated single page application using netlify or alike, how much work would be needed to  decouple the generation of the web assets (css, js, ...) and the associated api backend server ?

**Comment by jourdain** [comment]
While __trame__ really target a niche usage where you need a server process for each client to properly handle large data processing and/or rendering interactively. I could see a usage of it for building reporting where the processing happen once and the generate ui+state get saved so it can be looked at offline without any dedicated server later on. 

I guess I would see 3 types of websites that could be generated with trame:

1. Basic application with no real dynamic user input (computed values, interactive 2D charts, graphs...)
   - for such use case the update needed for trame to support it would be trivial (Less than a day)
2. Same as 1 + local rendering with some vtk scene while interaction is limited to camera manipulation
   - This one is bit more complex as we would need to also dump the various mesh arrays and swap client implementation to fetch them via http rather than ws. (A couple of days)
3. Fully feature interactive application with sliders and server side computing
   - This is not going to happen. This is where trame design shine and could not be managed with a standard http server.

HTH,

Seb
